packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -376,5 +400,39 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function targetNodeFunctionAncestorNodeHasVoidReturnType(
+      node: TSESTree.Node,
+    ): boolean {
+      const targets: [
+        AST_NODE_TYPES.FunctionDeclaration,
+        AST_NODE_TYPES.ArrowFunctionExpression,
+        AST_NODE_TYPES.FunctionExpression,
+      ] = [
+        AST_NODE_TYPES.FunctionDeclaration,
+        AST_NODE_TYPES.ArrowFunctionExpression,
+        AST_NODE_TYPES.FunctionExpression,
+      ];
+
+      if (!node.parent) {
+        return false;
+      }
+
+      if (
+        targets[0] === node.parent.type ||
+        targets[1] === node.parent.type ||
+        targets[2] === node.parent.type
+      ) {
```

Comment:
```suggestion
      if (
        AST_NODE_TYPES.FunctionDeclaration === node.parent.type ||
        AST_NODE_TYPES.ArrowFunctionExpression === node.parent.type ||
        AST_NODE_TYPES.FunctionExpression === node.parent.type
      ) {
```

[Refactor] Why do we need the `targets` tuple if we don't iterate over it? IMO if we directly compare `node.parent.type` to AST node types, it will be more readable. What do you think?

Reply:
Comparing variables directly seems much more readable!